src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -539,20 +540,24 @@ private CompilationTracker GetCompilationTracker(ProjectId projectId)
 
         private Solution AddProject(ProjectId projectId, ProjectState projectState)
```

Comment:
does this already contain all files/references added? I remember we did every in sequence before. like add a project, and a file one by one, add a reference one by one. and change project configuration one by one and etc.

did we change that? otherwise, the scope of global operation seems too small